backends/graphics/gles-custom/gles-custom.cpp

```diff
@@ -0,0 +1,703 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+// Needed so we don't get errors when including the Raspberry Pi headers
+#define FORBIDDEN_SYMBOL_ALLOW_ALL
+#include "backends/graphics/gles-custom/gles-custom.h"
+#include "common/textconsole.h"
+#include "common/config-manager.h"
+#ifdef USE_OSD
+#include "common/translation.h"
+#endif
+
+#include <GLES/gl.h>
+#include <EGL/egl.h>
+
+#ifdef USE_GLES_RPI
+#include <bcm_host.h>
+DISPMANX_ELEMENT_HANDLE_T dispman_element;
+DISPMANX_DISPLAY_HANDLE_T dispman_display;
+DISPMANX_UPDATE_HANDLE_T dispman_update;
+static EGL_DISPMANX_WINDOW_T nativewindow;
+#endif
+
+#ifdef USE_GLES_FBDEV
+// Includes for framebuffer size retrieval
+#include <sys/ioctl.h>
+#include <linux/fb.h>
+#include <linux/vt.h>
+#include <fcntl.h>
+#include <unistd.h>
+struct fbdev_window nativewindow;
+#endif
+
+#ifdef USE_GLES_KMS
+// This hacky define is needed since there is an struct member called "virtual" in xf86drm.h
+#define virtual __virtual
+#include <xf86drm.h>
+#undef virtual
+
+#include <xf86drmMode.h>
+#include <gbm.h>
+
+static struct {
+	struct gbm_device *dev;
+	struct gbm_surface *surface;
+} gbm;
+
+static struct {
+	int fd;
+	drmModeModeInfo *mode;
+	uint32_t crtc_id;
+	uint32_t connector_id;
+} drm;
+
+struct drm_fb {
+	struct gbm_bo *bo;
+	uint32_t fb_id;
+};
+
+struct gbm_bo *bo;
+struct drm_fb *fb;
+fd_set fds;
+
+void page_flip_handler(int fd, unsigned int frame,
+		  unsigned int sec, unsigned int usec, void *data)
+{
+	int *waiting_for_flip = (int*)data;
+	*waiting_for_flip = 0;
+}
+
+static drmEventContext evctx;
+
+void drm_fb_destroy_callback(struct gbm_bo *bob, void *data);
+
+struct drm_fb * drm_fb_get_from_bo(struct gbm_bo *bob)
+{
```

Comment:
Same thing with wrong formatting, now even pointer formatting is incorrect. Also why are you using C syntax, not C++?